Source/monster.cpp

```diff
@@ -2341,7 +2341,7 @@ void OverlordAi(int i)
 	int v = GenerateRnd(100);
 	if (abs(mx) >= 2 || abs(my) >= 2) {
 		if ((monster._mVar2 > 20 && v < 4 * monster._mint + 20)
-		    || ((monster._mVar1 == MM_WALK || monster._mVar1 == MM_WALK2 || monster._mVar1 == MM_WALK3)
+		    || ((monster._mVar1 == static_cast<int>(MonsterMode::MoveNorthwards) || monster._mVar1 == static_cast<int>(MonsterMode::MoveSouthwards) || monster._mVar1 == static_cast<int>(MonsterMode::MoveSideways))

```

Comment:
```suggestion
		    || (IsAnyOf(static_cast<MonsterMode>(monster._mVar1), MonsterMode::MoveNorthwards, MonsterMode::MoveSouthwards, MonsterMode::MoveSideways)
```
Cuts down on the explicit casts, and I reckon promoting `monster._mVar1` to `MonsterMode` is preferable to casting away from the enum type.

As a side note this check is repeated multiple times for the different AI procedures, the checks for `Var1 is a walking mode && Var2 == 0` would be useful as a member function of `MonsterStruct`. It seems to be used to check if the monster was walking before it started idling.

Reply:
Extracting a function to encapsulate that is out of scope for this. Though I agree, I'd have to check these monster vars in more detail to understand why var2 is being checked as well. Also, we'd have to decide on what to call it.

The `IsAnyOf` change has been implemented.

Reply:
I suppose `oldMode` would follow the original convention, but `previousMode` sounds better.